frontend/lib/src/components/widgets/ButtonGroup/ButtonGroup.tsx

```diff
@@ -325,7 +336,13 @@ function ButtonGroup(props: Readonly<Props>): ReactElement {
           Root: {
             style: {
               flexWrap: "wrap",
-              gap: gap,
+              columnGap: getButtonGroupColumnGap(style, theme),
+              rowGap: rowGap,
+              maxWidth: "fit-content",
+              "::after": {
+                content: "''",
+                flex: 10000,
+              },
```

Comment:
nit: Should we maybe set this only in the Segment-style. I wondered why the pills snapshots changed, and this might explain it 🙂 I didn't see a visual difference so it might be fine to leave it, but perhaps it makes sense to narrow down the scope so that it is clearer in which case this is relevant. Also it would make sense to leave a comment explaining what this does, something like `// adding an empty pseudo-element after the last button in the group. This will make buttons only as big as needed without stretching to the whole container width (aka let them 'hug' to the side)`

Reply:
Ah good call, I didn't intend to change this for pills as well. Also good idea to add an explanation